frappe/integrations/utils.py

```diff
@@ -35,21 +35,47 @@ def make_post_request(url, **kwargs):
 def make_put_request(url, **kwargs):
 	return make_request('PUT', url, **kwargs)
 
-def create_request_log(data, integration_type, service_name, name=None, error=None):
-	if isinstance(data, str):
-		data = json.loads(data)
-
-	if isinstance(error, str):
-		error = json.loads(error)
+def create_request_log(
+	data,
+	integration_type=None,
+	service_name=None,
+	name=None,
+	error=None,
+	request_headers=None,
+	output=None,
+	**kwargs
+):
+	"""
+		DEPRECATED: The parameter integration_type will be removed in the next major release.
+		Use is_remote_request instead.
+	"""
+
+	get_json = lambda obj: obj if isinstance(obj, str) else frappe.as_json(obj, indent=1)
+
+	if integration_type == "Remote":
+		kwargs["is_remote_request"] = 1
+
+	elif integration_type == "Subscription Notification":
+		kwargs["request_description"] = integration_type
+
+	reference_doctype = reference_docname = None
+	if "reference_doctype" not in kwargs:
+		if isinstance(data, str):
+			data = json.loads(data)
+
+		reference_doctype = data.get("reference_doctype")
+		reference_docname = data.get("reference_docname")
```

Comment:
Note for reviewer: maybe we should look into deprecating this behaviour as well. Currently used in some payment integrations.